src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -61,25 +61,27 @@ public class Session {
             }
         }
 
-        internal Session(SessionManager manager, IIdentity user, string id, Interpreter interpreter, string commandLineArguments) {
+        internal Session(SessionManager manager, IIdentity user, string id, Interpreter interpreter, string commandLineArguments, ILogger<Session> sessionLogger) {
```

Comment:
Just `ILogger`. The `<T>` thing has no semantic meaning, it's just a trick to make dependency injection create different loggers for you automatically. So if the component doesn't obtain its logger from dependency injection, it should just use `ILogger`.